tests/validation/tests/v3_api/test_cluster_templates.py

```diff
@@ -0,0 +1,448 @@
+    assert cluster.appliedSpec.rancherKubernetesEngineConfig.network.plugin ==\
+        "flannel"
+
+    # Verify flannel pod in the kube-system namespace
+    cmd = "get pods -l k8s-app=flannel --namespace kube-system"
+    pod_result = execute_kubectl_cmd(cmd)
+
+    assert (len(["items"])) == 1
+
+    for pod in pod_result["items"]:
+        print(pod["metadata"]["name"])
+        assert "flannel" in (pod["metadata"]["name"])
+
+    # Perform Backup
+    backup = cluster.backupEtcd()
+    backupname = backup['metadata']['name']
+    etcdbackups = cluster.etcdBackups(name=backupname)
+    etcdbackupdata = etcdbackups['data']
+    backupId = etcdbackupdata[0]['id']
+    print("BackupId", backupId)
+    wait_for_backup_to_active(standard_user_client, cluster, backupname)
+
+    cluster_cleanup(standard_user_client, cluster)
+
+
+def test_cluster_template_create_edit_adminuser():
+
+    client = get_admin_client()
+    cluster_template_create_edit(client)
+
+
+def test_cluster_template_create_edit_stduser():
+
+    standard_user_client = Client(
+                         url=CATTLE_API_URL,
+                         token=user_token["user_standard"]["token"],
+                         verify=False)
+    cluster_template_create_edit(standard_user_client)
+
+
+def test_cluster_template_add_owner():
+
+    cluster_config1 = get_cluster_config(RANCHER_DEFAULT_K8S_VERSION)
+    client = get_admin_client()
+
+    standard_newuser_client = Client(
+                                url=CATTLE_API_URL,
+                                token=user_token["newuser_standard"]["token"],
+                                verify=False)
+
+    # As an Admin, create a cluster template, revision and update the members
+    # list with the new user as owner
+    template_name = random_test_name("template")
+    cluster_template = client.create_cluster_template(
+        name=template_name, description="test-template")
+    print(cluster_template)
+
+    principalid = user_token["newuser_standard"]["user"]["principalIds"]
+    members = [{
+        "type": "member",
+        "accessType": "owner",
+        "userPrincipalId": principalid
+    }]
+
+    cluster_template = client.update(cluster_template,
+                                     name=template_name,
+                                     members=members)
+    # As an owner of the template create a revision using the template
+    # and also create a cluster using the template revision
+    revision_name = random_test_name("revision1")
+    cluster_template_revision = \
+        standard_newuser_client.create_cluster_template_revision(
+            name=revision_name,
+            clusterConfig=cluster_config1,
+            clusterTemplateId=cluster_template.id)
+    cluster_name = "test-auto"
+    cluster = create_node_cluster(
+        standard_newuser_client, name=cluster_name,
+        clusterTemplateRevisionId=cluster_template_revision.id)
+
```

Comment:
Create another CT with 1 CTR and make sure that "standard_newuser_client" is not able to create new revisions for this CT or deploy a cluster with existing CTR.